### PR TITLE
[DEV-8435] Fix indexes and improve deprecation language and text placement.

### PR DESCRIPTION
**Pull request by todddunagan-bah**
**Description:**
High level description of what the PR addresses should be put here. Should be detailed enough to communicate to a PO what this PR addresses without diving into the technical nuances

**Technical details:**
The technical details for the knowledge of other developers. Any detailed caveats or specific deployment steps should be outlined here.

**Requirements for PR merge:**

1. [x] Unit & integration tests updated
2. [x] API documentation updated
3. [ ] Necessary PR reviewers:
    - [x] Backend
    - [ ] Frontend <OPTIONAL>
    - [ ] Operations <OPTIONAL>
    - [ ] Domain Expert <OPTIONAL>
4. [x] Matview impact assessment completed
5. [x] Frontend impact assessment completed
6. [x] Data validation completed
7. [x] Appropriate Operations ticket(s) created
8. [ ] Jira Ticket [DEV-123](https://federal-spending-transparency.atlassian.net/browse/DEV-123):
    - [ ] Link to this Pull-Request
    - [ ] Performance evaluation of affected (API | Script | Download)
    - [ ] Before / After data comparison

**Area for explaining above N/A when needed:**
```
```

### CODE REVIEW

**Review by sethstoudenmier — DISMISSED** [review]
Nice touch for the sake of the API docs! Tested sort changes locally and all checks out.

Screen Shot 2022-02-22 at 10 37 39 AM

File: usaspending_api/recipient/migrations/0013_create_uei_indexes_for_performance.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Generated by Django 2.2.18 on 2021-03-15 20:28
```
Comment (sethstoudenmier) [inline usaspending_api/recipient/migrations/0013_create_uei_indexes_for_performance.py:1]: Probably left over from a copy and paste, but this is no longer true. Could look to just add these migration steps into the `recipient 0012` migration. At the very least, comment should be updated.

File: usaspending_api/recipient/models.py [hunk 2]
```diff
@@ -120,11 +120,8 @@ class Meta:
         #         public.recipient_profile USING btree (recipient_name DESC NULLS LAST)
         #     SQL: CREATE INDEX recipient_profile_recipient_unique_id_d ON
         #         public.recipient_profile USING btree (recipient_unique_id DESC NULLS LAST)
```
Comment (sethstoudenmier) [inline usaspending_api/recipient/models.py:122]: Can you update this comment above to also mention the new UEI indexes that are manually created with RunSQL?

**Review by sethstoudenmier — CHANGES_REQUESTED** [review]
Minor requested changes around updating comments and possibly combining migrations.

File: usaspending_api/recipient/migrations/0013_create_uei_indexes_for_performance.py (hunk 1 above)
Comment (sethstoudenmier) [reply usaspending_api/recipient/migrations/0013_create_uei_indexes_for_performance.py:1]: @todddunagan-bah if you could address this as well

Comment (todddunagan-bah) [reply usaspending_api/recipient/migrations/0013_create_uei_indexes_for_performance.py]: My mistake. Done.

File: usaspending_api/recipient/models.py (hunk 2 above)
Comment (todddunagan-bah) [reply usaspending_api/recipient/models.py]: Done.

**Review by sethstoudenmier — APPROVED** [review]

File: usaspending_api/recipient/models.py [hunk 3]
```diff
@@ -99,32 +99,24 @@ class Meta:
...
-        #         public.recipient_profile USING btree (recipient_name DESC NULLS LAST)
-        #     SQL: CREATE INDEX recipient_profile_recipient_unique_id_d ON
-        #         public.recipient_profile USING btree (recipient_unique_id DESC NULLS LAST)
+        #     MIGRATION: 0006_auto_20210315_2028 (2 indexes)
+        #     MIGRATION: 0013_create_uei_indexes_for_performance (1 index)
         #
+
         indexes = [
             GinIndex(fields=["award_types"]),
-            models.Index(fields=["recipient_unique_id"]),
```
Comment (todddunagan-bah) [inline usaspending_api/recipient/models.py:126]: For the historical reader's sake: removing the DUNS index list item here as it's superfluous to the index created by the column definition. Should offer no functional change, just housekeeping.

**Pull request merged**